File: FirServer/FirServer/Manager/ModelManager.cs
using System;
using System.Collections.Generic;
using log4net;
using FirServer.Model;

namespace FirServer.Manager
{
    public class ModelManager : BaseManager
    {
        private static Dictionary<string, BaseModel> models = new Dictionary<string, BaseModel>();
        private static readonly ILog logger = LogManager.GetLogger(AppServer.repository.Name, typeof(ModelManager));

        public ModelManager()
        {
        }

        public override void Initialize()
        {
            models.Clear();
        }

        public BaseModel GetModel(string strKey)
        {
            if (models.ContainsKey(strKey))
            {
                return models[strKey];
            }
            return null;
        }

        public void AddModel(string strKey, BaseModel model)
        {
            if (models.ContainsKey(strKey))
            {
                throw new Exception();
            }
            models.Add(strKey, model);
        }

        public void RemoveModel(string strKey)
        {
            if (models.ContainsKey(strKey))
            {
                models.Remove(strKey);
            }
        }
    }
}

File: FirServer/FirServer/src/com/manager/ModelManager.java
package com.manager;

import java.util.LinkedHashMap;
import java.util.Map;

import com.interfaces.IModel;

public class ModelManager extends BaseManager {
    private static Map<String, IModel> models = new LinkedHashMap<String, IModel>();

    @Override
    public void Initialize() {
    }

    public void AddModel(String typeName, IModel model) {
        if (!models.containsKey(typeName)) {
            models.put(typeName, model);
        }
    }

    public IModel GetModel(String typeName) {
        IModel model = null;
        if (models.containsKey(typeName))
        {
            model = models.get(typeName);
        }
        return model;
    }

    public void RemoveModel(String typeName) {
        if (models.containsKey(typeName)) {
            models.remove(typeName);
        }
    }

    @Override
    public void OnDispose() {
    }
}
